dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/CDIContainerImpl.java

```diff
@@ -636,6 +638,15 @@ private ExtensionArchive newSPIExtensionArchive(ServiceReference<CDIExtensionMet
         Set<Class<?>> beanClasses = webSphereCDIExtensionMetaData.getBeanClasses();
         Set<Class<? extends Annotation>> beanDefiningAnnotationClasses = webSphereCDIExtensionMetaData.getBeanDefiningAnnotationClasses();
 
+        if (! beanDefiningAnnotationClasses.isEmpty()) {
+            if (! com.ibm.ws.kernel.productinfo.ProductInfo.getBetaEdition()) {
+                throw new UnsupportedOperationException("This method is beta and is not available.");
```

Comment:
I think that in this case we probably don't want to throw an exception. We should just ignore the content... as if it is not there.

Otherwise, any components which are adding this will also need to have their own beta guards.

Reply:
we decided to leave this exception and force implementors to include beta guards ... this makes it more obvious when it does not behave as may be expected